openpathsampling/tests/analysis/utils.py

```diff
@@ -0,0 +1,500 @@
+    scheme : :class:`.MoveScheme`
+        the move scheme used in the mock simulation
+    ensemble : :class:`.Ensemble` or None
+        the ensemble to shoot from. If ``None``, a random mover is selected
+        from the movers in ``group_name``.
+    accepted : bool or None
+        if ``True`` or ``False``, force the acceptance to by true or false,
+        respectively. If ``None`` (default) use the internal logic to
+        determine acceptance. NOTE: This only bypasses the Monte Carlo
+        proposal probability calculation. Moves that are lot legally
+        possible to accept will raise an AnalysisTestSetupError if you try
+        to force them to be accepted.
+    group_name : str
+        the name of the groups to select movers from (default
+        ``'shooting'``).
+    """
+
+    def __init__(self, shooting_index, partial_traj, scheme, ensemble=None,
+                 accepted=None, group_name='shooting'):
+        super(MockBackwardShooting, self).__init__(
+            shooting_index=shooting_index,
+            partial_traj=partial_traj,
+            direction='backward',
+            scheme=scheme,
+            ensemble=ensemble,
+            accepted=accepted,
+            group_name=group_name
+        )
+
+
+class MockRepex(MockMove):
+    """Mock a replica exchange move.
+
+    Parameters
+    ----------
+    scheme : :class:`.MoveScheme`
+        the move scheme used in the mock simulation
+    ensembles : List[:class:`.Ensemble`] or None
+        The input ensembles for the replica exchange move. If ``None``, a
+        random mover is selected.
+    group_name : str
+        the name of the groups to select movers from (default
+        ``'repex'``).
+    """
+    def __init__(self, scheme, ensembles=None, group_name='repex'):
+        super(MockRepex, self).__init__(scheme=scheme,
+                                        ensembles=ensembles,
+                                        group_name=group_name)
+
+
+class MockPathReversal(_MockSingleEnsembleMove):
+    """Mock a path reversal  move.
+
+    Parameters
+    ----------
+    scheme : :class:`.MoveScheme`
+        the move scheme used in the mock simulation
+    ensemble : List[:class:`.Ensemble`] or None
+        The input ensemble for the path reversal move. If ``None``, a
+        random mover is selected.
+    group_name : str
+        the name of the groups to select movers from (default
+        ``'pathreversal'``).
+    """
+    def __init__(self, scheme, ensemble=None, group_name='pathreversal'):
+        super(MockPathReversal, self).__init__(scheme=scheme,
+                                               ensemble=ensemble,
+                                               group_name=group_name)
+
+
+def _do_single_step(init_conds, move, org_by_group):
+    change = move(init_conds)
+    if org_by_group:
+        change = move.wrap_org_by_group(change, init_conds)
+    return change
+
+def run_moves(init_conds, moves, org_by_group=True):
```

Comment:
expected 2 blank lines
```suggestion


def run_moves(init_conds, moves, org_by_group=True):
```